bumble/hci.py

```diff
@@ -1460,15 +1463,55 @@ def init_from_bytes(hci_object, data, offset, fields):
     def dict_from_bytes(data, offset, fields):
         result = collections.OrderedDict()
         for (field_name, field_type) in fields:
+            variadic_field = False
             # The field_type may be a dictionary with a mapper, parser, and/or size
             if isinstance(field_type, dict):
                 if 'size' in field_type:
                     field_type = field_type['size']
                 elif 'parser' in field_type:
                     field_type = field_type['parser']
+                elif 'size_field' in field_type:
```

Comment:
I'd like to propose an alternative way of dealing with field arrays, in a way that uses the same conventions for constructor parameters and field names as currently (see PR #230)